### ISSUE TITLE
Layer's zip file has unexpected top folder when using compose

### ISSUE THREAD

**Issue by danielvijge** [serverless/serverless#12868]
### Issue description

(mentioned earlier in https://github.com/serverless/serverless/issues/12845#issuecomment-2407379437 after that issue was already closed. Not sure if it will be seen, so creating a new issue for it)

When packaging a layer using compose the folder names in the zip file are not correct.

I have the following folder structure:

```
-- layer
  \-- ext-dependencies
    \-- java
      \-- lib
        \-- {many *.jar files}
  -- serverless.yml
```
In serverless.yml path is set to ext-dependencies.

When doing the packaging from compose the files in the zip file have a top level folder called `encies`.
So extracting the zip file has this folder structure:
```
-- encies
  \-- java
    \-- lib
      \-- {many *.jar files}
```
When doing the packaging from within the layers folder (so not using compose), the zip file has this folder structure:
```
-- java
  \-- lib
    \-- {many *.jar files}
```
Because the jar files are zip'ed in an additional top level folder, the lambda functions cannot find it in the layer, and all functions fail to execute.

### Context

Tested on both 4.4.5 (which contains the fix for #12845) and 4.4.6.

**Comment by czubocha** [comment]
Hey @danielvijge, thanks for reporting this. We'll include the fix in the next release.

**Comment by czubocha** [comment]
Hey @danielvijge, the fix has been released in version [4.4.7](https://github.com/serverless/serverless/releases/tag/v4.4.7). Feel free to let us know if you encounter any further issues.
